src/Response/Struct/Office.php

```diff
@@ -0,0 +1,92 @@
+    /** @var string[] */
+    public array $phones;
+    /** @var string[] */
+    public array $emails;
+    public OfficeAddress $address;
+    public string $info;
+    public string $currency;
+    public ?string $language;
+    public OpeningHours $openingHours;
+    public ?OpeningHours $halfDayOpeningHours = null;
+    /** @var ShipmentType[] */
+    public array $shipmentTypes;
+    public bool $cardPaymentAllowed = true;
+    public string $partnerCode;
+    public string $hubCode;
+    public string $hubName;
+    public string $hubNameEn;
+    public OfficeType $officeType;
+
+    public static function fromArray(array $officeData): self
+    {
+        Assert::integer($officeData['id']);
+        Assert::stringNotEmpty($officeData['code']);
+        Assert::boolean($officeData['isMPS']);
+        Assert::boolean($officeData['isAPS']);
+        Assert::stringNotEmpty($officeData['name']);
+        Assert::stringNotEmpty($officeData['nameEn']);
+        Assert::allString($officeData['phones']);
+        Assert::allString($officeData['emails']);
+        Assert::string($officeData['info']);
+        Assert::stringNotEmpty($officeData['currency']);
+        Assert::nullOrString($officeData['language']);
+        Assert::integer($officeData['normalBusinessHoursFrom']);
+        Assert::integer($officeData['normalBusinessHoursTo']);
+        Assert::nullOrInteger($officeData['halfDayBusinessHoursFrom']);
+        Assert::nullOrInteger($officeData['halfDayBusinessHoursTo']);
+        Assert::string($officeData['partnerCode']);
+        Assert::stringNotEmpty($officeData['hubCode']);
+        Assert::stringNotEmpty($officeData['hubName']);
+        Assert::stringNotEmpty($officeData['hubNameEn']);
+
+        $office = new self();
+        $office->id = $officeData['id'];
+        $office->code = $officeData['code'];
+        $office->isMPS = $officeData['isMPS'];
+        $office->isAPS = $officeData['isAPS'];
+        $office->name = $officeData['name'];
+        $office->nameEn = $officeData['nameEn'];
+        $office->phones = $officeData['phones'];
+        $office->emails = $officeData['emails'];
+        $office->address = OfficeAddress::fromArray($officeData['address']);
+        $office->info = $officeData['info'];
+        $office->currency = $officeData['currency'];
+        $office->language = $officeData['language'];
+        $office->openingHours = new OpeningHours(
+            $officeData['normalBusinessHoursFrom'],
+            $officeData['normalBusinessHoursTo']
+        );
+        if (null !== $officeData['halfDayBusinessHoursFrom'] && null !== $officeData['halfDayBusinessHoursTo']) {
+            $office->halfDayOpeningHours = new OpeningHours(
+                $officeData['halfDayBusinessHoursFrom'],
+                $officeData['halfDayBusinessHoursTo']
+            );
+        }
+        $office->shipmentTypes = array_map(fn ($type) => ShipmentType::get($type), $officeData['shipmentTypes']);
```

Comment:
Food for thought: with DPD we started ignoring enums like this one we couldn't map (https://github.com/answear/dpd-pl-pickup-services-bundle/pull/5). Maybe it'd be a good idea here as well?